### ISSUE TITLE
[Elastic Agent] Running custom inputs under Elastic Agent

### ISSUE THREAD

**Issue by ruflin** [elastic/beats#30019]
Elastic Agent is a supervisor and can run any binary supporting the [elastic-agent-client](https://github.com/elastic/elastic-agent-client). This allows Elastic Agent to be extended with additional binaries. At the moment these binaries must be built by Elastic for signing. One of the ideas discussed is, that at one point this can also be used by non Elastic binaries.

But the above still puts a burden on running custom inputs / collectors built by users. The collectors might already exist and adding support for the elastic-agent-client is not feasible. Instead these collectors / scripts could be executed by an input similar to the old [exectbeat](https://github.com/christiangalsterer/execbeat).

This exec input could be configured to run any command on a predefined schedule and would read the data from stdout. Different output formats like prometheus, json and others would be supported.

The configuration could look something like:

```
inputs:
  - type: command
    schedule: 10s
    format: json
    run: /foo/bar/my-collector.py
```

This would run `my-collector.py` every 10s.

There are security concerns around being able to just execute commands in an Elastic Agent especially in the context of Fleet. I would expect this input to be in a separate binary potentially not shipped by default. In addition, it could be blocked in `capabilities.yml` by default. A user would have to enable it for every Elastic Agent where this should be available.

**Comment by ruflin** [comment]
Lots of previous technical discussions on this happened already in https://github.com/elastic/beats/issues/18323

**Comment by jlind23** [comment]
@ruflin could you keep only one issue to track the work down?  :-)

**Comment by ruflin** [comment]
I don't think it is the same issue. The previous one is around Beats, this one here is for Elastic Agent. But the security challenges are the same. My assumption with Elastic Agent around is that it is less likely that this will still happen in Beats so we could close the other one but make sure we look up the technical discussion there.

**Comment by jlind23** [comment]
@kvch could you consider it for the input manager proposal?

**Comment by ericleong86** [comment]
+1 to this.

From security perspective, most established EDR vendors have ways to run custom scripts on adhoc or upon certain event basis (some even have remote access capabilities). Of course, all of these functionality comes with privilege management on the console and proper audit reporting. 

The use case is mainly to be used as incident response tool when a breach is detected. Some customers leverage this function as additional forensics collection after the infected endpoint/server has been isolated (quarantined).
